### PR TITLE
Setup Eslint for search

### LINKED ISSUE #3160: Configure ESLint for `search`
`src/api/search` needs to be configured to add the new `@senecacdot/eslint-config-telescope` to `devDependencies` and configure `ESLint` to actually work.

### PR DESCRIPTION
**Pull request by JiaHua-Zou**
Fixes #3160 
## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Steps to test the PR
pnpm install
pnpm lint

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by tcvan0707 — APPROVED** [review]

File: src/api/search/package.json [hunk 1]
```diff
@@ -21,6 +24,9 @@
     "node": ">=12.0.0"
   },
   "devDependencies": {
-    "nodemon": "2.0.15"
+    "@senecacdot/eslint-config-telescope": "1.1.0",
```
Comment (aserputov) [inline src/api/search/package.json:27]: We need to delete supertest.

Comment (JiaHua-Zou) [reply src/api/search/package.json:27]: The `Eslint` is complaining that we need to put `supertest` in the devDependencies.

**Review by tcvan0707 — APPROVED** [review]

File: src/api/search/package.json (hunk 1 above)
Comment (TueeNguyen) [reply src/api/search/package.json:27]: I think whatever from `jest` we use like `nock`, `jest-fetch-mock`, we can add to devDependencies. Eslint complained to me about it too

Comment (JiaHua-Zou) [reply src/api/search/package.json:27]: so eslint does not like it when the `supertest` and `nock` are not listed in package.json. We can either add it to the `devDependencies` or add a ignore line right on top of it when we request it.

Comment (TueeNguyen) [reply src/api/search/package.json:27]: Personally, I think it's ok to add it to package.json

**Review by aserputov — APPROVED** [review]

**Review by cindyledev — APPROVED** [review]

**Pull request merged**